src/microbe_stage/AgentProjectile.cs

```diff
@@ -7,22 +8,39 @@
 [SceneLoadedClass("res://src/microbe_stage/AgentProjectile.tscn", UsesEarlyResolve = false)]
 public class AgentProjectile : RigidBody, ITimedLife
 {
+    private Particles particles;
+
+    [JsonProperty]
+    private float? FadeTimeRemaining { get; set; }
+
     public float TimeToLiveRemaining { get; set; }
     public float Amount { get; set; }
     public AgentProperties Properties { get; set; }
     public Node Emitter { get; set; }
 
     public void OnTimeOver()
     {
-        Destroy();
+        BeginDestroy();
     }
 
     public override void _Ready()
     {
+        particles = GetNode<Particles>("Particles");
+
         AddCollisionExceptionWith(Emitter);
         Connect("body_entered", this, "OnBodyEntered");
     }
 
+    public override void _Process(float delta)
+    {
+        if (FadeTimeRemaining == null)
+            return;
+
+        FadeTimeRemaining -= delta;
+        if (FadeTimeRemaining < MathUtils.EPSILON)
```

Comment:
Actually now with the nullcheck, this could be replaced just by `FadeTimeRemaning <= 0` as there is no danger of overshooting the value anymore.